File: content/java-rs/upcoming/source/includes/specify-documents-to-return.java
# start-limit-method
FindPublisher<Document> findPublisher = restaurants.find(
                eq("cuisine", "Italian")).limit(5);

Flux.from(findPublisher)
        .doOnNext(x -> System.out.println(x.getString("name")))
        .blockLast();
# end-limit-method

# start-sort-method
FindPublisher<Document> findPublisher = restaurants.find(
                eq("cuisine", "Italian")).sort(ascending("name"));

Flux.from(findPublisher)
        .doOnNext(x -> System.out.println(x.getString("name")))
        .blockLast();
# end-sort-method

# start-skip
FindPublisher<Document> findPublisher = restaurants.find(
                eq("borough", "Manhattan")).skip(10);

Flux.from(findPublisher)
        .doOnNext(x -> System.out.println(x.getString("name")))
        .blockLast();
# end-skip

# start-limit-sort-skip
FindPublisher<Document> findPublisher = restaurants.find(
                eq("cuisine", "Italian"))
        .sort(ascending("name"))
        .limit(5)
        .skip(10);

Flux.from(findPublisher)
        .doOnNext(x -> System.out.println(x.getString("name")))
        .blockLast();
# end-limit-sort-skip

File: source/includes/read/specify-documents-to-return.kt
package org.example
import com.mongodb.ConnectionString
import com.mongodb.MongoClientSettings
import com.mongodb.client.model.Filters.eq
import com.mongodb.client.model.Sorts
import com.mongodb.kotlin.client.MongoClient

// start-data-class
data class Restaurant(
    val name: String,
    val borough: String,
    val cuisine: String
)
// end-data-class

fun main() {
    val uri = "<connection string URI>"

    val settings = MongoClientSettings.builder()
        .applyConnectionString(ConnectionString(uri))
        .retryWrites(true)
        .build()

    val mongoClient = MongoClient.create(settings)
    val database = mongoClient.getDatabase("sample_restaurants")
    val collection = database.getCollection<Restaurant>("restaurants")

    // start-limit
    val results = collection
        .find(eq(Restaurant::cuisine.name, "Italian"))
        .limit(5)

    results.forEach { result ->
        println(result)
    }
    // end-limit

    // start-sort
    val results = collection
        .find(eq(Restaurant::cuisine.name, "Italian"))
        .sort(Sorts.ascending(Restaurant::name.name))

    results.forEach { result ->
        println(result)
    }
    // end-sort

    // start-skip
    val results = collection
        .find(eq(Restaurant::cuisine.name, "Italian"))
        .skip(10)

    results.forEach { result ->
        println(result)
    }
    // end-skip

    // start-limit-sort-skip
    val results = collection
        .find(eq(Restaurant::cuisine.name, "Italian"))
        .sort(Sorts.ascending(Restaurant::name.name))
        .skip(10)
        .limit(5)

    results.forEach { result ->
        println(result)
    }
    // end-limit-sort-skip
}
